q2cli/builtin/tools.py

```diff
@@ -907,3 +908,274 @@ def cache_status(cache):
     success = "Status of the cache at the path '%s':\n\n%s" % \
         (cache, output)
     click.echo(CONFIG.cfg_style('success', success))
+
+
+replay_in_fp_help = (
+    'filepath to a QIIME 2 Archive (.qza or .qzv) or directory of Archives'
+)
+replay_recurse_help = (
+    'if in-fp is a directory, will also search sub-directories when finding '
+    'Archives to parse'
+)
+replay_validate_checksums_help = (
+    'check that replayed archives are intact and uncorrupted'
+)
+replay_parse_metadata_help = (
+    'parse the original metadata captured in provenance for review or replay'
+)
+replay_use_recorded_metadata_help = (
+    're-use the original metadata captured in provenance'
+)
+replay_suppress_header_help = (
+   'do not write header/footer blocks in the output script'
+)
+replay_verbose_help = (
+    'print status messages to stdout while processing'
+)
+replay_dump_recorded_metadata_help = (
+    'write the original metadata captured in provenance to disk in the '
+    '--metadata-out-fp directory'
+)
+replay_out_fp_help = (
+    'the filepath where your replay script should be written'
+)
+
+
+@tools.command(name='replay-provenance', cls=ToolCommand)
+@click.option('--in-fp', required=True, help=replay_in_fp_help)
+@click.option('--recurse/--no-recurse',
+              default=False,
+              show_default=True,
+              help=replay_recurse_help)
+@click.option('--usage-driver',
+              default='cli',
+              show_default=True,
+              help='the target interface for your replay script',
+              type=click.Choice(['python3', 'cli'], case_sensitive=False))
+@click.option('--validate-checksums/--no-validate-checksums',
+              default=True,
+              show_default=True,
+              help=replay_validate_checksums_help)
+@click.option('--parse-metadata/--no-parse-metadata',
+              default=True,
+              show_default=True,
+              help=replay_parse_metadata_help)
+@click.option('--use-recorded-metadata/--no-use-recorded-metadata',
+              default=False,
+              show_default=True,
+              help=replay_use_recorded_metadata_help)
+@click.option('--suppress-header/--no-suppress-header',
+              default=False,
+              show_default=True,
+              help=replay_suppress_header_help)
+@click.option('--verbose/--no-verbose',
+              default=True,
+              show_default=True,
+              help=replay_verbose_help)
+@click.option('--dump-recorded-metadata/--no-dump-recorded-metadata',
+              default=True,
+              show_default=True,
+              help=replay_dump_recorded_metadata_help)
+@click.option('--metadata-out-fp',
```

Comment:
This parameter might be more accurately named as `--metadata-out-dir` so that it's clear that this output is written to a directory path (while `--out-fp` is just written to a file path).